addons/lunch/models/lunch_supplier.py

```diff
@@ -201,6 +201,8 @@ def write(self, values):
             self.env['lunch.order'].search([('supplier_id', 'in', self.ids)]).write({'company_id': values['company_id']})
         super().write(values)
         if not CRON_DEPENDS.isdisjoint(values):
+            # flush automatic_email_time field to call _sql_constraints
+            self.flush_model(['automatic_email_time'])
```

Comment:
would not it make sense to do something like 
```suggestion
if 'automatic_email_time' in values:
    self.flush_model(['automatic_email_time'])
```